controller/k8s/api.go

```diff
@@ -210,17 +237,19 @@ func NewAPI(
 		case Node:
 			api.node = sharedInformers.Core().V1().Nodes()
 			api.syncChecks = append(api.syncChecks, api.node.Informer().HasSynced)
+		case Secret:
+			api.secret = sharedInformers.Core().V1().Secrets()
+			api.syncChecks = append(api.syncChecks, api.secret.Informer().HasSynced)
 		}
 	}
-
 	return api
 }
 
 // Sync waits for all informers to be synced.
-func (api *API) Sync() {
-	api.sharedInformers.Start(nil)
-	api.spSharedInformers.Start(nil)
-	api.tsSharedInformers.Start(nil)
+func (api *API) Sync(stopCh chan struct{}) {
```

Comment:
The signature is more clear here if you mark this as a receive-only channel: `stopCh <-chan struct{}`.

This clarifies that this channel is used by the caller to invoke a stop, rather than for the callee to inform the caller of a stop.